docs/application-connector/docs/08-04-register-secured-api.md

```diff
@@ -51,3 +53,54 @@ This is an example of the `api` section of the request body for an API secured w
             },
         }  
 ```
+
+## Register a client certificate-secured API
+
+To register an API and secure it with client certificates, you must add the `credentials.certificateGen` object to the `api` section of the service registration request body. The Application Registry generates a ready to use certificate and key pair for every API registered this way. You can use the generated pair or replace it with your own certificate and key.
+
+Include this field in the service registration request body:
+
+| Field   |  Description |
+|:----------:|:------|
+| **commonName** |  Name of the generated certificate. Set as the `CN` field of the certificate Subject.  |
+
+This is an example of the `api` section of the request body for an API secured with generated client certificates:
+
+```
+    "api": {
+        "targetUrl": "https://sampleapi.targeturl/v1",
+        "credentials": {
+            "certificateGen": {
+                "commonName": "{CERT_NAME}"
+            },
+        }  
+```
+
+>**NOTE:** If you update the registered API and change the `certificateGen.commonName`, the Application Registry generates a new certificate-key pair for that API. When you delete an API secured with generated client certificates, the Application Registry deletes the corresponding certificate and key.
+
+### Details
+
+When you register an API with the `credentials.certificateGen` object, the Application Registry generates a SHA256withRSA-encrypted certificate and matching key. To enable communication between Kyma and an API secured with this authentication method, set the certificate as a valid authentication medium for all calls coming from Kyma in your external solution.
+
+The certificate and key pair is stored in a Secret in the `kyma-integration` Namespace. List all Secrets and find the one created for your API:
+
+```
+kubectl -n kyma-integration get secrets
+```
+
+To fetch the certificate and key encoded with base64, run this command:
+
+```
+kubectl -n kyma-integration get secrets app-{APP_NAME}-{SERVICE_ID}
+```
+
+>**NOTE:** Replace the `APP_NAME` placeholder with the name of the Application used to connect the external solution that is the origin of the API. Replace the `SERVICE_ID` placeholder with the ID of the registered service to which the API belongs. You get this ID after you register an external solution's service in Kyma.
+
+
+If the generated certificate doesn't meet your security standards or specific needs, you can use a custom certificate-key pair for authentication. To replace the generated certificate and key with a custom pair in Kyma, run this command:
```

Comment:
Please start that sentence with,
`... If you already have a client certificate with a key provided by your API...`
We need to highlight that this use case is supported.